### ISSUE TITLE
VoiceOver rotor on messages 

### ISSUE THREAD

**Issue by sempruijs** [vector-im/element-x-ios#660]
## problem

VoiceOver users navigate allot with the rotor.
With the rotor, you can select secondary options of the element that is selected. For visual people, you can see it like a right mouse click.

Currently, this is not implemented on messages, which is frustrating because VoiceOver users are used to navigate this way. (messengers like WhatsApp, iMessage and signal already have this and VoiceOver users love this feature)

## solution

Implement the rotor on messages. With 1 swipe they can delete messages, reply, forward, etc.

## what should happen

1. Turn VoiceOver on
2. View a conversation (so you can compose a message that view)
3. put the VoiceOver cursor on a message. (That is already send by you or someone else)
4. Swipe vertically to select secondary options. VoiceOver users hear options like: Reply, Remove, forward.
5. When the users hears the option that they want to select, they select it by double tapping. 
6. The action is performed.

**Describe alternatives you've considered**

One option is to keep it as it is. Currently the workflow to delete a message for instance is holding the cursor on a message that they want to delete, simulate a long press by triple tapping and clicking on the delete button.

For VoiceOver users that know allot about the subject, they know the triple tap exist. But an average VoiceOver user doesn't know this and feel frustrated that they cannot delete a message or reply to a message.

An average VoiceOver user does know about the rotor and when they revalidate the tutors teach about the rotor and not about the triple tap. So the rotor it is!

## links

[explenation of what the rotor is and how to use it](https://support.apple.com/en-us/HT204783)

[swiftui documentation for implementing secondary actions for swiftUI](https://developer.apple.com/documentation/swiftui/view/accessibilityactions(_:))

**Comment by kittykat** [comment]
@sempruijs Would you mind reaching out to me over Matrix to chat about the issues you've filed? You can reach me at @kittykat:matrix.org

**Comment by kittykat** [comment]
Each item in the bottom sheet should correspond to an item in the rotor. For example, for a bottom sheet containing:
```
reply
delete
reply in thread
react
forward
```

The rotor should offer:
```
[message bubble rotor actions] {
reply
delete
reply in thread
react
forward
}
```

**Comment by stefanceriu** [comment]
Hi, thanks for raising this. Unfortunately, the current APIs don't allow accessibility actions to be computed on the fly when highlighting a particular item and require them available upfront. That would mean computing them for every message displayed in the timeline which is too computationally expensive.

As a workaround we decided to expose on single `Message actions` to the rotor which should at least inform the users that there are other actions available. 

Another option was to just open the action menu on double tap but we decided against it as it would be redundant. What do you think?
